packages/babel-plugin-proposal-export-default-from/src/index.js

```diff
@@ -35,6 +35,11 @@ export default declare(api => {
 
         path.replaceWithMultiple(nodes);
       },
+      ImportDefaultSpecifier(path) {
+        if (!path.scope.hasOwnBinding(path.node.local.name)) {
+          path.scope.registerDeclaration(path.parentPath);
```

Comment:
I would prefer to do this as soon as the import declaration is created, in the `ExportNamedDeclaration` visitor. This makes it clearer _why_ we are doing it, and it prevents other plugin from running between the creating the declaration and registering it.

`path.replaceWithMultiple(nodes);` returns an array containing the new inserted paths: the first one should be the import declaration, and you can get it's default specifier.

Reply:
> `path.replaceWithMultiple(nodes)` returns an array containing the new inserted paths.

Yes it is. It seems that `@types/babel__traverse` declares out-of-dated signature [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/babel__traverse/index.d.ts#L274).